### ISSUE TITLE
GitLab OIDC integration group claims not working

### ISSUE THREAD

**Issue by shivpathak** [goharbor/harbor#11808]
**What can we help you?**

I have deployed harbor version **v1.10.2** in kubernetes and integrated with GitLab (GitLab as an OIDC provider), Authentication works fine.

My problem is gitlab returns **groups** claim in _/oauth/userinfo_ endpoint not in _ID Token_.
You may refer the table and last line over [gitlab-openid-connect](https://gitlab.com/gitlab-org/gitlab-foss/-/blob/master/doc/integration/openid_connect_provider.md).

GitLab's Groups/Project authorization is not working in Harbor with OIDC integration.

In harbor core pod I see following logs - 

```
2020-04-29T09:37:45Z [WARNING] [/common/utils/oidc/helper.go:331]: Unable to get groups from claims, claims: map[aud:a5f88de7d50760a8089ce3675b87bb61ff497458d633c4249886b34897cc9e22 auth_time:1.588148693e+09 exp:1.588153185e+09 iat:1.588153065e+09 iss:https://gitlab.com sub:3254751 sub_legacy:1d8428b1e6510c723bac46ba4a47fe53833c68d72b9834bde7289413e71e5e88], groups claims key: groups
```

**How OIDC configuration looks like in harbor**
<img width="526" alt="Screen Shot 2020-04-29 at 4 08 15 PM" src="https://user-images.githubusercontent.com/10459864/80587172-b736ab80-8a33-11ea-9cf1-5a3ae1522fb0.png">

**Please let me know in case any further information is needed?**

**Comment by shivpathak** [comment]
IMHO: harbor should read claims form _/oauth/userinfo_ endpoint.

**Comment by reasonerjt** [comment]
@shivpathak I checked the code actually it DOES read claims from userinfo endpoint as long as it's in the response of well-known url:
https://github.com/goharbor/harbor/blob/release-1.10.0/src/common/utils/oidc/helper.go#L248

Please double check that the userinfo endpoint is indeed containing the `groups` in the response.

**Comment by shivpathak** [comment]
@reasonerjt I just verified, you are correct it works fine with group claim, (I was testing wrongly)

What I was doing wrong?

After login with oidc user, I found that groups got created in harbor which the oidc user belongs in GitLab, which proofs that claim is working fine.

I was in wrong assumptions that the repository from git will be auto synced, and hence after OIDC user login I was looking inside projects option and thinking why GitLab projects didn't appear in projects menu. Of course I was wrong.

Closing the ticket here as groups claim works fine with GitLab with the same config mentioned in issue description.

Sorry for confusion and Thanks a lot for your response.
